internal/cloud/backend.go

```diff
@@ -465,13 +465,15 @@ func (b *Cloud) setConfigurationFields(obj cty.Value) tfdiags.Diagnostics {
 }
 
 func reconcileWorkspaceMappingEnvVars(w *WorkspaceMapping) tfdiags.Diagnostic {
-	// See: https://github.com/opentofu/opentofu/issues/814
-	if v := os.Getenv("TF_WORKSPACE"); v != "" && w.Name == "" {
-		if len(w.Tags) > 0 && !workspaceInTags(w.Tags, v) {
-			return invalidWorkspaceConfigMisconfigurationEnvVar
+	if v := os.Getenv("TF_WORKSPACE"); v != "" {
+		if w.Name != "" && w.Name != v {
```

Comment:
Found this missing case while running tests - if the user used `TF_WORKSPACE` and has `workspaces: { name = xx }` in config, to maintain compatibility, we need to validate that they have the same value or send an error.